third_party/parquet.patch

```diff
@@ -0,0 +1,7719 @@
+diff -ru -p1 --new-file src/parquet/parquet_types.cpp src/parquet/parquet_types.cpp
```

Comment:
was this file intentional to add?

Reply:
nvm, I see it is applied as part of the parquet build

Reply:
Thanks @BryanCutler. The patch file actually just added two files:
```
src/parquet/parquet_types.cpp
src/parquet/parquet_types.h
```

Those two files are thrift type files. The files are generated automatically. This is also a one time effort for each version and will not change when regenerate. However, it need both `flex` and `bison` to generate. I tried to build `flex` and `bison` through bazel but realized that they are more complicated than thrift itself. So I think maybe we could just build those two thrift files offline (and not to build flex and bison to generate those two files in compile time).

Also, I could not find an option in Bazel to support "override" or "additional extra" file on top of the `http_archive` to add those two files in the build. So I use a patch.

Reply:
Ah ok, sounds good. Thanks for the explanation!

Reply:
Would it be useful to document how to build this patch, either as a comment block or a separate "build instructions" file? It might be useful if we want to refresh this patch down the road.

Reply:
Also, do we want to check in the thrift type files used to generate this patch?